lib/modules/dosomething/dosomething_api/resources/reportback_item_resource.inc

```diff
@@ -158,7 +158,58 @@ function _reportback_item_resource_index($campaigns, $exclude, $status, $count,
 
   if (variable_get('rogue_collection', FALSE)) {
     $client = dosomething_rogue_client('v1');
-    $response = $client->getAllReportbacks($inputs);
+
+    // Get the query params to see if we are populating client or server side gallery portion.
+    // If page param is present, we are populating "view more" section of gallery (client side).
+    $query_parts = explode("&", $_SERVER['REQUEST_URI']);
+    $query_params = [];
+
+    // Separate the base url from the first query param.
+    $first_query_param = explode("?", $query_parts);
+    array_push($query_params, $first_query_param[1]);
+
+    // Add the remaining query params into array.
+    $remaining_query_params = array_shift($query_parts);
+    array_push($query_params, $remaining_query_params);
+
+    // Extract the exclude value for the exclude param.
+    // This excludes first 6 reportbacks that have already been displayed.
+    foreach ($query_params as $param) {
+      $param_parts = explode('=', $param);
+      if ($param_parts[0] === 'exclude') {
+        $exclude = end($param_parts);
+      }
+    }
+
+    // Check to see if there is a page param. If so, add page param in request to Rogue.
+    $last_query_param = end($query_parts);
+    $last_query_param_parts = explode("=", $last_query_param);
+
+    if ($last_query_param_parts[0] === 'page') {
+      $page_number = end($last_query_param_parts);
+
+      if ($page_number) {
+        $response = $client->getAllReportbacks([
+          'filter' => [
+            'campaign_id' => $campaigns,
+            'exclude' => $exclude,
+          ],
+          'as_user' => dosomething_user_get_field('field_northstar_id'),
```

Comment:
Not sure if I'm quite understanding what you're saying here either! Isn't this in the query string of the request?